From: Dana Jansens <danakj@orodu.net>
Date: Fri, 5 Aug 2011 16:05:10 +0000 (-0400)
Subject: Load the rc.xml config file given on --config-file the same as Openbox (use the whole ...
X-Git-Url: http://git.openbox.org/?p=dana%2Fobconf.git;a=commitdiff_plain;h=cc7a18807663313ef111d86a75844ded0416a889;hp=9cffa6a9ddfc4074f3de8d0302404d70c2818d8f

Load the rc.xml config file given on --config-file the same as Openbox (use the whole path given).
---

diff --git a/src/main.c b/src/main.c
index 9035e8d..a576253 100644
--- a/src/main.c
+++ b/src/main.c
@@ -233,11 +233,10 @@ int main(int argc, char **argv)
     }
 
     xmlIndentTreeOutput = 1;
-    if (!obt_xml_load_config_file(parse_i,
-                                  "openbox",
-                                  (obc_config_file ?
-                                   obc_config_file : "rc.xml"),
-                                  "openbox_config"))
+    if (!((obc_config_file &&
+           obt_xml_load_file(parse_i, obc_config_file, "openbox_config")) ||
+          obt_xml_load_config_file(parse_i, "openbox", "rc.xml",
+                                   "openbox_config")))
     {
         obconf_error(_("Failed to load an rc.xml. You have probably failed to install Openbox properly."), TRUE);
         exit_with_error = TRUE;

